### PR TITLE
Fixed issue in collection of Constructor and Driver Standings

### PR DESCRIPTION
**Pull request by y34hbuddy**
Small PR to fix an issue with querying Constructor Standings and Driver Standings

### CODE REVIEW

**Review by Samuel-Roach — CHANGES_REQUESTED** [review]
Hey, thanks for the PR, it's great to see people invested in the project!

Although I can see the reasoning behind this change, I'm going to go into why this isn't quite right, and how to fix the issue :D

The issue (I think) you're trying to fix can be replicated when a user calls the following:
```py
mercedes = e.constructor('mercedes').get_constructor_standings()
print(mercedes)
```
which will currently output something along the lines of
```
[ConstructorStanding(position=0, position_text=, points=0.0, wins=0, constructor=Constructor(constructor_id=, url=, name=, nationality=)), ConstructorStanding(position=0, position_text=, points=0.0, wins=0, constructor=Constructor(constructor_id=, url=, name=, nationality=)), ...]
```
This is obviously incorrect, and is caused by the incorrect attempted parsing of the API. In `Requester.get_driver_standings()` we return the JSON array of StandingsLists, which we then attempt to coerce into a `constructor_standings` in `Ergast.get_driver_standings()` etc. (square blocks don't fit in triangle holes!!).

Whilst your solution would solve this problem, by making sure we're trying to create a `ConstructorStanding` from the correct data, we'd experience dataloss. When querying the API we want to ensure that we gather all the relevant data that it provides. Whilst your offered fix solves the issue, it removes any possibility of grabbing more data from the API. For instance if we want to grab all the mercedes constructor standings, we will only get the 0th position of their StandingsList (and we probably want to make sure we also grab the seasons where they win ;) ).

> Sidenote: This is kinda stemmed from the fact that the ergast API chooses to nest their standings in StandingsTable->StandingsList

The best solution for this, in my opinion, is to either make sure we return a StandingsList for the singular gets, or a [StandingsList] when we want to gather more than one.

To make this change, I'd recommend we change [line 284](https://github.com/Samuel-Roach/ergast-py/blob/master/ergast_py/ergast.py#L284) in `ergast.py` to construct a standings_lists instead, which will correctly parse the data that it has received, and not drop any data from the API.

An example in the `Ergast.get_driver_standings()` function would be:
```python
def get_driver_standings(self) -> list[StandingsList]:
        """
        Get a list of driver standings from the current query
        """
        return self._get_items(self.requester.get_driver_standings,
                               self.type_constructor.construct_standings_lists)
```

> It may be worth updating the docstring on the changed functions to recall what they actuall represent

**Comment by y34hbuddy** [comment]
This totally makes sense... let me take another stab at it!

File: ergast_py/ergast.py [hunk 1]
```diff
@@ -409,7 +409,7 @@ def get_constructor_standings(self) -> list[StandingsList]:
         Get a list of constructor standings from the current query
         """
         return self._get_items(self.requester.get_constructor_standings,
-                               self.type_constructor.construct_constructor_standings)
+                               self.type_constructor.construct_standings_lists)
```
Comment (Samuel-Roach) [inline ergast_py/ergast.py:412]: What about if we run
```python
mercedes = e.constructor('mercedes').season().get_constructor_standing()
print(mercedes)
```

Currently, we'll still print
```
ConstructorStanding(position=0, position_text=, points=0.0, wins=0, constructor=Constructor(constructor_id=, url=, name=, nationality=))
```
so we should probably update the singular versions of these functions to use `construct_standings_lists` too!

**Comment by y34hbuddy** [comment]
Cheers, @Samuel-Roach -- thanks for letting me contribute! And go, McLaren!

**Comment by y34hbuddy** [comment]
Hi @Samuel-Roach are you planning any releases soon?

As a bit of background, I'm building an F1 Integration (by way of your library) for Home Assistant (https://www.home-assistant.io/ / https://github.com/home-assistant/core) so that users (like me) can do nifty things like set the lights for race time, or play a celebratory sound when their favourite driver overtakes Sir Lewis in the Championship.

I'm pretty much ready to submit my PR to Home Assistant, but can't do until I have a working `ergast_py` in PyPI :)

**Comment by Samuel-Roach** [comment]
Hi @y34hbuddy,

Great to see this being put to use! Yeah I had intended to add a couple more changes before the next release but the need for it sooner rather than later makes sense, my changes can wait! 

I'll try and get the release published tomorrow for you :)

**Comment by Samuel-Roach** [comment]
Version 0.7.0 has been released. Available at https://pypi.org/project/ergast-py/

**Comment by y34hbuddy** [comment]
Hi @Samuel-Roach - if you're keen to see where this ended up: https://github.com/home-assistant/core/pull/80326

https://github.com/y34hbuddy/core/blob/62d068c501a086a3e0911429a9ef04ec5b2b0e00/homeassistant/components/formula_one/__init__.py#L71

**Pull request merged**